### PR TITLE
Add `try_spawn` function for Anvil and Geth bindings

### PR DESCRIPTION
**Pull request by moricho**
## Motivation
Closes: https://github.com/alloy-rs/alloy/issues/136

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/node-bindings/src/anvil.rs [hunk 1]
```diff
@@ -313,6 +346,107 @@ impl Anvil {
             chain_id: self.chain_id.or(chain_id),
         }
     }
+
+    /// Consumes the builder and spawns `anvil`. If spawning fails, returns an error.
+    #[track_caller]
```
Comment (mattsse) [inline crates/node-bindings/src/anvil.rs:351]:
```suggestion
```

**Review by mattsse — CHANGES_REQUESTED** [review]
this is a great start, we can now simplify the existing spawn functions by calling `try_spawn.expect`

File: crates/node-bindings/src/anvil.rs [hunk 2]
```diff
@@ -65,6 +66,38 @@ impl Drop for AnvilInstance {
     }
 }
 
+/// Errors that can occur when working with the [`Anvil`].
+#[derive(Debug, Error)]
+pub enum AnvilError {
+    /// Spawning the anvil process failed.
+    #[error("couldnt start anvil")]
+    Spawn,
```
[the remark this answers is not in the document]
Comment (moricho) [reply crates/node-bindings/src/anvil.rs]: fixed at: https://github.com/alloy-rs/alloy/pull/226/commits/22168d69ae6f95cfb4402b3c2063e2b639cf99e2

File: crates/node-bindings/src/anvil.rs (hunk 1 above)
Comment (moricho) [reply crates/node-bindings/src/anvil.rs]: fixed at: https://github.com/alloy-rs/alloy/pull/226/commits/263d850eef02d0fb92518324d8b20c9676caa939

**Comment by moricho** [comment]
@mattsse 
Thanks for the review! 
I have tried to categorize some of the errors related to Geth. However, I am unsure if I have left anything out or if the categorization is accurate. If you have any suggestions or advice, I would appreciate your input.
https://github.com/alloy-rs/alloy/pull/226/files#diff-edd3384dd7db8fd08c3dfd953104f384359ab2f8211cdae1504eebd93ffbf38fR174-R210

**Review by mattsse — CHANGES_REQUESTED** [review]
sorry for the delay, 
error variants look fine,
last thing left to do is to change the `fn spawn` to 

```rust
fn spawn(&self) -> Instance {
   self.try_spawn().unwrap()
}
```

**Comment by moricho** [comment]
Added the above change in https://github.com/alloy-rs/alloy/pull/226/commits/037a15f15cb9b350258869a6be263850721b6228

**Review by mattsse — APPROVED** [review]

**Pull request merged**